wfl/Database/IDBManager.cs

```diff
@@ -7,7 +7,9 @@ namespace Symphono.Wfl.Database
     public interface IDBManager
     {
         Task InsertRestaurantAsync(RestaurantDto r);
+        Task<bool> CheckRestaurantIdAsync(string id);
         Task<IEnumerable<RestaurantDto>> GetAllRestaurantsAsync();
+        Task<RestaurantDto> GetRestaurantWithIdAsync(string Id);
```

Comment:
Makes sense to me. I think that using a RestaurantEntity to interact with the database might be better. Maybe something we could address in a service layer PR.